controllers/prep_handlers.go

```diff
@@ -185,20 +259,38 @@ func (r *ImageBasedUpgradeReconciler) handlePrep(ctx context.Context, ibu *lcav1
 				return
 			}
 		} else if progress == "completed-stateroot" {
-			result, err = r.launchPullImages(ibu, progressfile)
+			result, err = r.launchPrecaching(ctx, imageListFile, progressfile)
 			if err != nil {
 				r.Log.Error(err, "Failed to launch get-seed-image phase")
 				return
 			}
+		} else if progress == "precaching-in-progress" {
+			result, err = r.queryPrecachingStatus(ctx, progressfile)
+			if err != nil {
+				r.Log.Error(err, "Failed to query get-seed-image phase")
+				return
+			}
 		} else if progress == "completed-precache" {
 			r.runCleanup(ibu)
 
+			// Fetch final precaching job report summary
+			conditionMessage := "Prep completed"
+			status, err := precache.QueryJobStatus(ctx, r.Client)
+			if err == nil && status.Message != "" {
+				conditionMessage += fmt.Sprintf(": %s", status.Message)
+			}
+
+			// Cleanup precaching resources
+			if err = precache.Cleanup(ctx, r.Client); err != nil {
```

Comment:
I would prefer to see the cleanup done in the handleAbort, so that the logs stick around if they are needed

Reply:
Pre-caching cleanup moved to handleAbort